src/table/components/virtualized-table/HeaderCell.tsx

```diff
@@ -9,36 +13,73 @@ interface HeaderCellProps {
   data: {
     columns: Column[];
     headerStyle: GeneratedStyling;
+    layout: TableLayout;
+    embed: stardust.Embed;
+    translator: ExtendedTranslator;
+    changeSortOrder: ChangeSortOrder;
   };
 }
 
+const StyledHeaderCell = styled(Box, {
+  shouldForwardProp: (prop: string) => prop !== 'borderWidth',
+})(({ borderWidth }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  borderStyle: 'solid',
+  borderWidth,
+  padding: '4px 12px',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  boxSizing: 'border-box',
+  '&&:hover': {
+    '& button, & svg': { opacity: 1 },
+  },
+  fontWeight: 'bold',
+}));
+
 const HeaderCell = ({ index, style, data }: HeaderCellProps) => {
-  const {
-    columns,
-    headerStyle: { sortLabelColor, ...applicableStyle },
-  } = data;
-  const datum = columns[index];
+  const { columns, headerStyle, embed, layout, translator, changeSortOrder } = data;
+  const { sortLabelColor, ...applicableStyle } = headerStyle;
+  const column = columns[index];
   const isLastColumn = columns.length - 1 === index;
+  const isCurrentColumnActive = layout.qHyperCube.qEffectiveInterColumnSortOrder[0] === column.colIdx;
+
+  const sortFromMenu = (evt: React.MouseEvent, newSortDirection: SortDirection) => {
+    evt.stopPropagation();
+    changeSortOrder(column, newSortDirection);
+  };
 
   return (
-    <div
+    <StyledHeaderCell
```

Comment:
Maybe Mui Grid could alternate this and below div element. no extra styles would required in that case i think